src/line_printer.cc

```diff
@@ -85,6 +85,7 @@ void LinePrinter::Print(string to_print, LineType type) {
     GetConsoleScreenBufferInfo(console_, &csbi);
 
     to_print = ElideMiddle(to_print, static_cast<size_t>(csbi.dwSize.X));
+    std::wstring w_to_print = Utf8ToWide(to_print);
```

Comment:
Another allocation for no reason. Down below, you're copying the guts out of this string. Why not just convert the string data directly into the destination buffer?

Reply:
I need to iterate over the results, so I'm not really sure on how the copying directly in to buffer would be used here?

Reply:
Yea that's fair. Good point.